src/metabase/sync/sync_metadata/tables.clj

```diff
@@ -94,34 +94,39 @@
               {:details
                (assoc (:details database) :version (:version db-metadata))}))
 
+(defn- cruft-dependent-columns [table-name]
+  ;; if this is a crufty table, mark initial sync as complete since we'll skip the subsequent sync steps
+  (let [is-crufty? (is-crufty-table? table-name)]
+    {:initial_sync_status (if is-crufty? "complete" "incomplete")
+     :visibility_type     (when is-crufty? :cruft)}))
+
+(defn create-table!
+  "Creates a new table in the database, ready to be synced.
+   Throws an exception if there is already a table with the same name, schema and database ID."
+  [database table]
+  (t2/insert-returning-instance!
+   Table
+   (merge (cruft-dependent-columns (:name table))
+          {:active                  true
+           :db_id                   (:id database)
+           :schema                  (:schema table)
+           :description             (:description table)
+           :database_require_filter (:database_require_filter table)
+           :display_name            (or (:display_name table) (humanization/name->human-readable-name table))
+           :name                    (:name table)})))
```

Comment:
I took the opportunity to extract this code out so the upload code could call this directly, rather than `create-or-reactivate-table!`.

Reply:
Even without the reuse it's easier to understand the code now 🙂